.github/workflows/build.yml

```diff
@@ -973,6 +973,50 @@ jobs:
           name: openbsd-${{ matrix.release }}-autotools-${{ matrix.EVENT_MATRIX }}-build
           path: .
 
+  apple-cmake-job:
+    runs-on: ${{ matrix.os }}
+    if: "!contains(github.event.head_commit.message, 'ci/macos skip') && !contains(github.event.head_commit.message, 'ci/macos/cmake skip')"
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest]
+        EVENT_MATRIX:
+          - IOS
+          - TVOS
+          - VISIONOS
+          - WATCHOS
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Cache Build
+        uses: actions/cache@v4
+        with:
+          path: build
+          key: ${{ matrix.os }}-cmake-${{ matrix.EVENT_MATRIX }}-v4
+
+      - name: Xcode
+        uses: maxim-lobanov/setup-xcode@v1
+        with:
+          xcode-version: latest-stable
+
+      - name: Build
+        shell: bash
+        run: |
+          if [ "${{ matrix.EVENT_MATRIX }}" == "IOS" ]; then
+            CMAKE_OSX_SYSROOT="iphoneos"
+          elif [ "${{ matrix.EVENT_MATRIX }}" == "TVOS" ]; then
+            CMAKE_OSX_SYSROOT="appletvos"
+          elif [ "${{ matrix.EVENT_MATRIX }}" == "VISIONOS" ]; then
+            CMAKE_OSX_SYSROOT="xros"
+          elif [ "${{ matrix.EVENT_MATRIX }}" == "WATCHOS" ]; then
+            CMAKE_OSX_SYSROOT="watchos"
+          fi
```

Comment:
This could be simplified by using 1-1 relation with the matrix.

Reply:
Agree. But you already merged the PR. :)

Reply:
Yep, since I want as much coverage as possible (with respect to a very limited CI time that libevent has)
But this can be done on top.

Reply:
#1643